### PR TITLE
PayPal: Remove Vault Listener Pattern

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

 - This PR migrates the `PayPalClient` vault flow away from the listener pattern.
 - Migration Guide updates will come in a later PR
 - Users must now balance calls to `PayPalWebCheckoutClient.vault()` with calls to `PayPalWebCheckoutClient.finishVault()` to complete a vault flow
 - This PR is a continuation of #307 

 ### Checklist

 - [x] Added a changelog entry

- @sshropshire

### CODE REVIEW

File: PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt [hunk 1]
```diff
@@ -69,31 +62,37 @@ class PayPalWebCheckoutClient internal constructor(
         analytics.notifyVaultStarted(request.setupTokenId)
         val result = payPalWebLauncher.launchPayPalWebVault(activity, request)
         when (result) {
-            is PayPalPresentAuthChallengeResult.Failure -> {
-                analytics.notifyVaultAuthChallengeFailed(request.setupTokenId)
-                vaultListener?.onPayPalWebVaultFailure(result.error)
-            }
-
             is PayPalPresentAuthChallengeResult.Success ->
                 analytics.notifyVaultAuthChallengeStarted(request.setupTokenId)
```
Comment (KunJeongPark) [inline PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt:66]: I think in iOS, we just have "challenge-required" event before launching a web view. 
So if we wanted to track success/failure of the entire 3ds flow including the web view launch, we would have a starting point. 

I guess Android equivalent is "challenge-received" so you just have an extra "start" on the auth.

**Review by KunJeongPark — APPROVED** [review]

File: PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt (hunk 1 above)
Comment (sshropshire) [reply PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt:66]: Yeah I'm hoping we'll be able to align analytics on iOS and Android sometime soon in an upcoming sprint.

**Comment by mikz88** [comment]
how do i get my paycheck?

**Review by aaronh197 — APPROVED** [review]

File: PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt (hunk 1 above)
Comment (aaronh197) [reply PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt:66]: https://github.com/paypal/paypal-android/commit/ff48f17bf502ed6c99ae4a9b1dd8caba998a7219

Comment (aaronh197) [reply PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt:66]: https://github.com/paypal/paypal-android/commit/ff48f17bf502ed6c99ae4a9b1dd8caba998a7219

**Pull request merged**